java/android/Portfile

```diff
@@ -107,10 +181,102 @@ platform macosx {
         }
     }
 }
-#
-#    post-install {
-#       system "( sleep 5 && while [ 1 ]; do sleep 1; echo y; done ) | }${prefix}/share/java/android-sdk-macosx/tools/android update sdk --no-ui"
-#    }
+
+variant platform_23 description "Install Android API 23 Version 6.0" {
+    build {
+        system "
+            export JAVA_HOME=\$(/usr/libexec/java_home --version 1.8)
+
+            echo '---> sdkmanager: Platforms 23'
+
+            pushd ${workpath}/android-sdk-macosx/tools/bin
+                ./sdkmanager                \
+                    'platforms;android-23'  \
+                    'sources;android-23'
+            popd
+        "
+    }
+}
+
+variant platform_24 description "Install Android API 24 Version 7.0" {
+    build {
+        system "
+            export JAVA_HOME=\$(/usr/libexec/java_home --version 1.8)
+
+            echo '---> sdkmanager: Platforms 24'
+
+            pushd ${workpath}/android-sdk-macosx/tools/bin
+                ./sdkmanager                \
+                    'platforms;android-24'  \
+                    'sources;android-24'
+            popd
+        "
+    }
+}
+
+variant platform_25 description "Install Android API 25 Version 7.1.2" {
+    build {
+        system "
+            export JAVA_HOME=\$(/usr/libexec/java_home --version 1.8)
+
+            echo '---> sdkmanager: Platforms 25'
+
+            pushd ${workpath}/android-sdk-macosx/tools/bin
+                ./sdkmanager                \
+                    'platforms;android-25'  \
+                    'sources;android-25'
+            popd
+        "
+    }
+}
+
+variant platform_26 description "Install Android API 26 Version 8.0" {
+    build {
+        system "
+            export JAVA_HOME=\$(/usr/libexec/java_home --version 1.8)
+
+            echo '---> sdkmanager: Platforms 26'
+
+            pushd ${workpath}/android-sdk-macosx/tools/bin
+                ./sdkmanager                \
+                    'platforms;android-26'  \
+                    'sources;android-26'
+            popd
+        "
+    }
+}
+
+variant platform_27 description "Install Android API 27 Version 8.1" {
+    build {
+        system "
+            export JAVA_HOME=\$(/usr/libexec/java_home --version 1.8)
+
+            echo '---> sdkmanager: Platforms 27'
+
+            pushd ${workpath}/android-sdk-macosx/tools/bin
+                ./sdkmanager                \
+                    'platforms;android-27'  \
+                    'sources;android-27'
+            popd
+        "
+    }
+}
+
+variant platform_28 description "Install Android API 28 Version 9.0" {
+    build {
+        system "
+            export JAVA_HOME=\$(/usr/libexec/java_home --version 1.8)
+
+            echo '---> sdkmanager: Platforms 28'
+
+            pushd ${workpath}/android-sdk-macosx/tools/bin
+                ./sdkmanager                \
+                    'platforms;android-28'  \
+                    'sources;android-28'
+            popd
+        "
+    }
```

Comment:
In general, variants should not be providing a different versions of the software. It is for features. Can these be made into sub-ports instead? Can they be installed simultaneously? If not, they can all be conflicting.

Reply:
Ditto.